home/export_assessments.py

```diff
@@ -0,0 +1,258 @@
+from openpyxl.worksheet.worksheet import Worksheet
+from wagtail.blocks import (  # type: ignore
+    StreamValue,
+    StructValue,  # type: ignore
+)
+from wagtail.query import PageQuerySet  # type: ignore
+
+
+@dataclass
+class ExportRow:
+    title: str = ""
+    page_id: int = 0
+    tags: str = ""
+    slug: str = ""
+    locale: str = ""
+    high_result_page: str = ""
+    high_inflection: str = ""
+    medium_result_page: str = ""
+    medium_inflection: str = ""
+    low_result_page: str = ""
+    low_inflection: str = ""
+    generic_error: str = ""
+    question_count: int = 0
+    questions: str = ""
+    error: str = ""
+    answers: str = ""
+    score: str = ""
+
+    @classmethod
+    def headings(cls) -> list[str]:
+        return [f.name for f in fields(cls)]
+
+    def to_dict(self) -> dict[str, str | int]:
+        return asdict(self)
+
+    def to_tuple(self) -> tuple[str | int, ...]:
+        return astuple(self)
+
+
+# Define your AssessmentExporter class
+class AssessmentExporter:
+    rows: list[dict[str | int, str | int]]
+
+    def __init__(self, queryset: PageQuerySet):
+        self.rows = []
+        self.queryset = queryset
+
+    def perform_export(self) -> list[dict[str | int, str | int]]:
+        for item in self.queryset:
+            questions_data = self.get_questions(item.questions)
+            for i, question_data in enumerate(questions_data):
+
+                self.rows.append(
+                    {
+                        "title": item.title,
+                        "page_id": item.id,
+                        "tags": self._comma_sep_qs(item.tags.all()),
+                        "slug": item.slug,
+                        "locale": str(item.locale),
+                        "high_result_page": str(item.high_result_page.slug),
+                        "high_inflection": item.high_inflection,
+                        "medium_result_page": str(item.medium_result_page.slug),
+                        "medium_inflection": item.medium_inflection,
+                        "low_result_page": str(item.low_result_page.slug),
+                        "low_inflection": 0,
+                        "generic_error": item.generic_error,
+                        "question_count": i + 1,
+                        **question_data,
+                    }
+                )
+        return self.rows
+
+    def get_questions(self, questions: StreamValue) -> list[StructValue]:
+        question_data = []
+        for question in questions:
+            answers = []
+            scores = []
+            for answer in question.value["answers"]:
+                answers.append(answer["answer"])
+                scores.append(str(answer["score"]))
+            question_data.append(
+                {
+                    "questions": (question.value["question"]).replace("\n", ","),
+                    "error": (question.value["error"]),
+                    "answers": ", ".join(answers),
```

Comment:
What if the answers have commas in them? We can probably use the `csv` library here, which will properly escape the commas, instead of a `", ".join`